v2/internal/controllers/insights_diagnosticsetting_crud_v20210501preview_test.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	insights "github.com/Azure/azure-service-operator/v2/api/insights/v1api20210501preview"
+	network "github.com/Azure/azure-service-operator/v2/api/network/v1api20201101"
+	storage "github.com/Azure/azure-service-operator/v2/api/storage/v1api20210401"
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/azure-service-operator/v2/internal/util/to"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+)
+
+func Test_Insights_DiagnosticSetting_CRUD(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+	rg_name := rg.Status.Name
+
+	// vnet := VMVirtualNetwork(tc, testcommon.AsOwner(rg))
+
+	vnet := &network.VirtualNetwork{
+		ObjectMeta: tc.MakeObjectMetaWithName(tc.Namer.GenerateName("vn")),
+		Spec: network.VirtualNetwork_Spec{
+			Owner:    testcommon.AsOwner(rg),
+			Location: tc.AzureRegion,
+			AddressSpace: &network.AddressSpace{
+				AddressPrefixes: []string{"10.0.0.0/16"},
+			},
+		},
+	}
+
+	tc.CreateResourceAndWait(vnet)
+
+	vnet_name := vnet.Status.Name
+
+	accessTier := storage.StorageAccountPropertiesCreateParameters_AccessTier_Hot
+	kind := storage.StorageAccount_Kind_Spec_StorageV2
+	sku := storage.SkuName_Standard_LRS
+	acct := &storage.StorageAccount{
+		ObjectMeta: tc.MakeObjectMetaWithName(tc.NoSpaceNamer.GenerateName("stor")),
+		Spec: storage.StorageAccount_Spec{
+			Location:              tc.AzureRegion,
+			AllowBlobPublicAccess: to.Ptr(false),
+			Owner:                 testcommon.AsOwner(rg),
+			Kind:                  &kind,
+			Sku: &storage.Sku{
+				Name: &sku,
+			},
+			// TODO: They mark this property as optional but actually it is required
+			AccessTier: &accessTier,
+		},
+	}
+
+	tc.CreateResourceAndWait(acct)
+
+	subARMID := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/virtualNetworks/%s", tc.AzureSubscription, *rg_name, *vnet_name)
+
+	vnet_diagnosticSettings := &insights.DiagnosticSetting{
+		ObjectMeta: tc.MakeObjectMeta("diagnosticsetting"),
+		Spec: insights.DiagnosticSetting_Spec{
+			Owner: &genruntime.ArbitraryOwnerReference{
+				ARMID: subARMID,
```

Comment:
You shouldn't need to use an ARMID to reference the VNet. ArbitraryOwnerReference allows referencing an owner within the cluster, and ASO will take care of creating things in the correct order.

There's a helper for this:

```
Owner: testcommon.AsOwner(vnet)
```

Reply:
Updated with the reference rather than ARMID.